### DISCUSSION TITLE
[question] What happens when record struct with explicit layout has members

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1219] [General]
``` F#
[<Struct; StructLayout(LayoutKind.Explicit)>]
        type Point2D= 
            {
                [<FieldOffsetAttribute(0)>]
                x : float32
                [<FieldOffsetAttribute(4)>]
                y : float32
            }
            with
            member pt.sum = pt.x + pt.y
```
does pt.sum computed when this record struct is computed therefore the size of the struct is actually 12 bytes instead of 8 bytes

**Comment by Tarmil** [comment]
Properties declared as `member this.Foo` are always computed on demand, so they don't affect the size of `this`.

Extra fields would be declared with `let` or `member val`; but those are forbidden in records.

**Reply by Xyncgas** [reply]
Thanks
